cellengine/resources/compensation.py

```diff
@@ -113,15 +113,18 @@ def apply(self, file: "FcsFile", inplace: bool = False, **kwargs):
             DataFrame: if ``inplace=True``, updates `FcsFile.events` for
                 the target FcsFile
         """
-        data = file.get_events(**kwargs, inplace=True, destination=None)
+        data: DataFrame = file.get_events(
+            **kwargs, inplace=True, destination=None
+        )  # type: ignore
 
```

Comment:
I think that this should take `events` as an arg instead of `fcs_file`, and return the comped events. Does that make more sense?

Reply:
I guess it could.  (  applyToEvents vs applyToFcsfile).  Its the "inplace" that threw me off because if it updates "inplace", then I expected it to start from the existing event df.

Reply:
Maybe do something like:
If `file._events` use those AND ideally assert that kwargs here are the same as the kwargs that were used when calling `file.get_events`. Requires storing the kwargs for the latter, which could actually be useful.
Else call `file.get_events()`

Reply:
Are the changes in 0332b79 and 9f7446f what you intended? These counteract the changes I made in ffff4f6, where I changed `.apply()` to accept an events dataframe; I'm not sure whether I like this better.

Reply:
Those commits don't exist anymore, which ones do you mean?